### PR TITLE
Fixed a bug where inception layers had 2 3x3 conv kernels instead of …

### PR DESCRIPTION
**Pull request by manjunaths**
…one 3x3 and 5x5 kernel.

### CODE REVIEW

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by martinwicke** [comment]
Jenkins, test this please (flakes)

**Review by sguada — CHANGES_REQUESTED** [review]
This would break the current checkpoint.

**Comment by manjunaths** [comment]
Please, Can I know what needs to be done next?

- Manju

**Comment by sguada** [comment]
Maybe make the change including an extra parameter that changes the behavior, but maintains the previous behavior by default.

**Comment by drpngx** [comment]
@manjunaths feel free to address.

**Comment by sguada** [comment]
@manjunaths you would need to add an extra parameter to the network that keeps the old behavior by default, and that can be changed when needed.

**Comment by manjunaths** [comment]
Hello @sguada ,

Thank you. I added a switch for this and left the original behavior as default. Can you review the code and see if it matches to the standards  or if anything more needs to be done ? I found one more issue addressing the same change issue #443. I think that can also be closed.

Thanks,
- Manju

**Comment by martinwicke** [comment]
@sguada can you take a look at the changes?

**Review by sguada — CHANGES_REQUESTED** [review]
The idea if that use_5x5 filter is an argument of the functions, so you can pass it when calling.
It also would need a lot more documentation, specially about checkpoint not be compatible.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by sguada** [comment]
The changes have not been done.

File: tensorflow/contrib/slim/python/slim/nets/inception_v1.py [hunk 1]
```diff
@@ -343,6 +350,10 @@ def inception_v1(inputs,
       scope, 'InceptionV1', [inputs, num_classes], reuse=reuse) as scope:
     with arg_scope(
         [layers_lib.batch_norm, layers_lib.dropout], is_training=is_training):
+      # If we want the same filters as in the original paper. The default is 3x3.
+      if use_5x5:
+        filter_size=[5, 5]
+        filter_size_str=str(filter_size[0])+'x'+str(filter_size[1])
       net, end_points = inception_v1_base(inputs, scope=scope)
```
Comment (sguada) [inline tensorflow/contrib/slim/python/slim/nets/inception_v1.py]: Pass use_5x5 to inception_v1_base too.

File: tensorflow/contrib/slim/python/slim/nets/inception_v1.py [hunk 2]
```diff
@@ -30,7 +30,9 @@
 from tensorflow.python.ops import variable_scope
 
 trunc_normal = lambda stddev: init_ops.truncated_normal_initializer(0.0, stddev)
-
+# Required to be able to switch between (3x3) and (5x5) as specified in the paper, Figure 2(b).
+filter_size = [3, 3]
```
Comment (sguada) [inline tensorflow/contrib/slim/python/slim/nets/inception_v1.py]: Remove it.

File: tensorflow/contrib/slim/python/slim/nets/inception_v1.py [hunk 3]
```diff
@@ -30,7 +30,9 @@
...
+filter_size_str=str(filter_size[0])+'x'+str(filter_size[1])
```
Comment (sguada) [inline tensorflow/contrib/slim/python/slim/nets/inception_v1.py]: transform into a function that given filter_size returns the string.

File: tensorflow/contrib/slim/python/slim/nets/inception_v1.py [hunk 4]
```diff
@@ -30,7 +30,9 @@
...
 
 def inception_v1_base(inputs, final_endpoint='Mixed_5c', scope='InceptionV1'):
```
Comment (sguada) [inline tensorflow/contrib/slim/python/slim/nets/inception_v1.py]: Could you pass branch_2_filter_size as an argument? and use it inside.

File: tensorflow/contrib/slim/python/slim/nets/inception_v1.py [hunk 5]
```diff
@@ -100,7 +102,7 @@ def inception_v1_base(inputs, final_endpoint='Mixed_5c', scope='InceptionV1'):
           with variable_scope.variable_scope('Branch_2'):
             branch_2 = layers.conv2d(net, 16, [1, 1], scope='Conv2d_0a_1x1')
             branch_2 = layers.conv2d(
-                branch_2, 32, [3, 3], scope='Conv2d_0b_3x3')
+                branch_2, 32, filter_size, scope='Conv2d_0b_'+filter_size_str)
```
Comment (sguada) [inline tensorflow/contrib/slim/python/slim/nets/inception_v1.py]: use branch_2_filter_size and filter_size_str(branch_2_filter_size)

**Review by sguada — CHANGES_REQUESTED** [review]

**Comment by vrv** [comment]
@manjunaths any updates?

**Comment by manjunaths** [comment]
Sorry, I have a personal emergency and am  unable to look at this for now.



On May 5, 2017 3:15 AM, "Vijay Vasudevan" <notifications@github.com> wrote:

@manjunaths <https://github.com/manjunaths> any updates?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/tensorflow/tensorflow/pull/8295#issuecomment-299317701>,
or mute the thread

**Comment by vrv** [comment]
I'm very sorry to hear that :(.  I hope things get better, and feel free to reopen a new PR if you are still able and interested.

**Pull request closed without merging**